graylog2-web-interface/src/views/logic/fieldactions/ChangeFieldType/IndexSetsTable.tsx

```diff
@@ -96,6 +107,16 @@ const IndexSetsTable = ({ field, setIndexSetSelection, fieldTypes, initialSelect
     setIndexSetSelection(newSelection);
   }, [setIndexSetSelection]);
 
+  const isEntitySelectable = useCallback((entity) => {
+    const indexSetId = entity.id;
+
+    return isIndexFieldTypeChangeAllowed(indexSets[indexSetId]);
+  }, [indexSets]);
+
+  useEffect(() => {
+    IndexSetsActions.list(false);
+  }, []);
+
```

Comment:
Usually we create a react-query hook in these cases, to get rid of the store at some point. We alrealy have similar hooks in the `components/indices/hook` and could add one there to fetch all indices.